### PR TITLE
Update layout_stride_fix with LWG feedback

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: layout_stride_fix/layout_stride_fix.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ toc: true
 
 # Revision History
 
+# Revision 1:
+
+- LWG just wants the default constructor to always do what was proposed for all-static extents case
```
Comment (mhoemmen) [inline layout_stride_fix/layout_stride_fix.md:23]: Optionally:

```suggestion
- LWG wants the default constructor to behave the same regardless of whether all the extents are static: that is, it should fill in the strides in the same way as `layout_right::mapping` would.
```

File: layout_stride_fix/layout_stride_fix.md [hunk 2]
```diff
@@ -125,6 +128,8 @@ Add at the beginning of subsection 24.7.3.4.7.3 [mdspan.layout.stride.cons] inse
 
 We believe that it is preferable to preserve default constructibility of `layout_stride` for all specializations of `layout_stride` to simplify a number of generic programming cases -- specifically, if layout policies are used directly in higher-level data structures for which the user wants to enable default constructibility.
 
+Based on feedback in LWG we leave out the defaulted constructor for cases with dynamic extents and just always default construct layout_stride as if it were a layout_right with the same default constructed extents.
```
Comment (mhoemmen) [inline layout_stride_fix/layout_stride_fix.md:131]:
```suggestion
Based on feedback in LWG we leave out the defaulted constructor for cases with dynamic extents and just always default construct `layout_stride::mapping` as if it were a `layout_right::mapping` with the same default constructed extents.
```

**Review by mhoemmen — APPROVED** [review]

File: layout_stride_fix/layout_stride_fix.md (hunk 2 above)
Comment (dalg24) [inline layout_stride_fix/layout_stride_fix.md:131]:
```suggestion
Based on feedback in LWG, we leave out the defaulted constructor for cases with dynamic extents and just always default construct `layout_stride` as if it were a `layout_right` with the same default constructed extents.
```

**Review by dalg24 — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

File: layout_stride_fix/layout_stride_fix.md [hunk 3]
```diff
@@ -138,18 +143,14 @@ In subsection 24.7.3.4.7.1 [mdspan.layout.stride.overview] replace:
...
 
 Add at the beginning of subsection 24.7.3.4.7.3 [mdspan.layout.stride.cons] insert:
 
 ```c++
-    constexpr mapping() 
-      requires(extents_type::rank_dynamic()==0 && extents_type::rank()>0) noexcept;
+    constexpr mapping() noexcept;
 ```
 
 *Effects:* Direct-non-list-initializes _`extents_`_ with `extents_type()` , and for all $d$ in the range $[$`0, `_`rank_`_$)$,
```
Comment (mhoemmen) [inline layout_stride_fix/layout_stride_fix.md:156]:
```
*Preconditions*: `layout_right::mapping<extents_type>{}.required_­span_­size()` is representable as a value of type `index_­type` ([[basic.fundamental]](https://eel.is/c++draft/basic.fundamental)).
```

**Review by mhoemmen — APPROVED** [review]

**Pull request merged**